lib/ref_arch_setup/install.rb

```diff
@@ -25,61 +31,307 @@ def initialize(target_master)
+        raise remote_error unless success
+      end
+
+      return tarball_path_on_master
+    end
+
+    # Copies the PE tarball from the specified location to the temp working directory
+    # on the target master
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] path The pe tarball path
+    #
+    # @return [true,false] Based on exit status of the bolt task
+    def copy_pe_tarball_on_target_master(path)
+      command = "cp #{path} #{TMP_WORK_DIR}"
+      success = BoltHelper.run_cmd_with_bolt(command, @target_master)
+      return success
+    end
+
+    # Handles the specified tarball path when the target master is not localhost
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] path The pe tarball path
+    #
+    # @return [true,false] Based on exit status of the bolt task
+    def handle_tarball_path_with_remote_target_master(path)
+      remote_flag = "#{@target_master}:"
+
+      if path.start_with?(remote_flag)
+        actual_path = path.sub!(remote_flag, "")
+        success = file_exist_on_target_master?(actual_path)
+        success = copy_pe_tarball_on_target_master(actual_path) if success
+      else
+        success = File.exist?(path)
+        success = upload_pe_tarball(path) if success
+      end
+
+      return success
+    end
+
+    # Handles the specified tarball path
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] path The PE tarball path
+    #
+    # TODO: improve "host to install on"? ("host where PE will be installed?")
+    #
+    # @return [string] The tarball path on the target master
+    def handle_tarball_path(path)
+      filename = File.basename(path)
+      tarball_path_on_master = "#{TMP_WORK_DIR}/#{filename}"
+      file_not_found_error = "File not found: #{path}"
+      upload_error = "Unable to upload tarball to the RAS working directory on #{@target_master}"
+      copy_error = "Unable to copy tarball to the RAS working directory on #{@target_master}"
+
+      if target_master_is_localhost?
+        raise file_not_found_error unless File.exist?(path)
+        error = copy_error
+        success = upload_pe_tarball(path)
+      else
+        error = upload_error
+        success = handle_tarball_path_with_remote_target_master(path)
+      end
+
+      raise error unless success
+
+      return tarball_path_on_master
+    end
+
+    # Handles the PE tarball based on the the path (URL / file)
+    # and target master (local / remote)
+    #
+    # TODO: Ensure it is valid once we have a reader/validator class
+    #
+    # @author Bill Claytor
+    #
+    # @param [string] pe_tarball Path to PE tarball file
+    #
+    # @return [string] The tarball path on the master after copying if successful
+    def handle_pe_tarball(pe_tarball)
+      error = "Unable to handle the specified PE tarball path: #{pe_tarball}"
+      validate_extension(pe_tarball)
+      tarball_path_on_master = if valid_url?(pe_tarball)
```

Comment:
I have a personal dislike for anything = if...   But that may be just me.  Did rubocop push you to do it this way, or was it just a choice?

Reply:
Leaving this based on discussion